test/common/upstream/health_check_fuzz.cc

```diff
@@ -221,6 +260,215 @@ void TcpHealthCheckFuzz::raiseEvent(const Network::ConnectionEvent& event_type,
+      grpc::health::v1::HealthCheckResponse::ServingStatus servingStatus =
+          convertToGrpcServingStatus(grpc_respond.grpc_respond_bytes().status());
+      std::vector<uint8_t> response = serializeResponse(servingStatus);
+      bufferList.push_back(response);
+      ENVOY_LOG_MISC(trace, "Responded with a serialized frame with status: {}",
+                     grpc_respond.grpc_respond_bytes().status());
+
+      uint64_t chunk_size = grpc_respond.grpc_respond_bytes().chunk_size_for_structured_response() %
+                            bufferList[0].size();
+      if (chunk_size == 0) {
+        ++chunk_size;
+      }
+      // Logically splitting a serialized response into fixed chunk sizes generated by the proto
+      for (size_t i = 0; i < bufferList[0].size() && !test_sessions_[0]->interval_timer_->enabled_;
+           i += chunk_size) {
+        bool last_chunk = false;
+        if (i > bufferList[0].size()) {
+          // The length of the last chunk
+          chunk_size = bufferList[0].size() - (i - chunk_size + 1);
+          last_chunk = true;
+        }
+        const auto data = std::make_unique<Buffer::OwnedImpl>(bufferList[0].data() + i, chunk_size);
+        ENVOY_LOG_MISC(trace, "Responded with structured data");
+        test_sessions_[0]->stream_response_callbacks_->decodeData(*data,
+                                                                  !has_trailers && last_chunk);
+      }
+      break;
+    }
+    case test::common::upstream::GrpcRespondBytes::kGrpcRespondUnstructuredBytes: {
+      constexpr auto max_chunks = 128;
+      for (int i = 0;
+           i <
+           std::min(
+               max_chunks,
+               grpc_respond.grpc_respond_bytes().grpc_respond_unstructured_bytes().data().size());
+           ++i) {
+        std::vector<uint8_t> chunk(
+            grpc_respond.grpc_respond_bytes().grpc_respond_unstructured_bytes().data(i).begin(),
+            grpc_respond.grpc_respond_bytes().grpc_respond_unstructured_bytes().data(i).end());
+        bufferList.push_back(chunk);
+      }
+      ENVOY_LOG_MISC(trace, "Responded with arbitrarily generated bytes which have no structure.");
+
+      // If the interval timer is enabled, that means that the rpc is complete, as decodeData hit a
+      // certain branch that called onRpcComplete(), logically representing a completed rpc call.
+      // Thus, skip the next responses until explicility invoking interval timer as cleanup.
+      for (size_t i = 0; i < bufferList.size() && !test_sessions_[0]->interval_timer_->enabled_;
+           ++i) {
+        bool end_stream_on_data = !has_trailers && i == bufferList.size() - 1;
+        const auto data =
+            std::make_unique<Buffer::OwnedImpl>(bufferList[i].data(), bufferList[i].size());
+        ENVOY_LOG_MISC(trace, "Responded with arbitrary data");
+        test_sessions_[0]->stream_response_callbacks_->decodeData(*data, end_stream_on_data);
+      }
+    }
+    default:
+      break;
```

Comment:
this is also unreachable i think

Reply:
Yup, good catch: I have a validate on the oneof clause. Changed to macro.